hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/AccountBalanceSigningUtils.java

```diff
@@ -0,0 +1,124 @@
+import java.security.SignatureException;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Utility class for signing account balance files
+ */
+public class AccountBalanceSigningUtils {
+
+    private static final Logger logger = LogManager.getLogger(AccountBalanceSigningUtils.class);
+    /**
+     * Hidden constructor
+     */
+    private AccountBalanceSigningUtils() {}
+
+
+
+    /**
+     * Sets up the constructable registry, and configures
+     * <p>
+     * Should be called before using stream utilities
+     * <p>
+     * Do <strong>NOT</strong> call this during standard node operation, as it will interfere with static settings
+     */
+    public static void initializeSystem() {
+        BootstrapUtils.setupConstructableRegistry();
+
+        // we don't want deserialization to fail based on any of these settings, not needed for services
+        //        SettingsCommon.maxTransactionCountPerEvent = Integer.MAX_VALUE;
+        //        SettingsCommon.maxTransactionBytesPerEvent = Integer.MAX_VALUE;
+        //        SettingsCommon.transactionMaxBytes = Integer.MAX_VALUE;
+        //        SettingsCommon.maxAddressSizeAllowed = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Generates a signature file for the account balance file
+     *
+     * @param signatureFileDestination the full path where the signature file will be generated
+     * @param streamFileToSign         the stream file to be signed
+     * @param keyPair                  the keyPair used for signing
+     * @return true if the signature file was generated successfully, false otherwise
+     */
+    public static boolean signAccountBalanceFile(
+            @NonNull final Path signatureFileDestination,
+            @NonNull final Path streamFileToSign,
+            @NonNull final KeyPair keyPair) {
+
+        Objects.requireNonNull(signatureFileDestination, "signatureFileDestination must not be null");
+        Objects.requireNonNull(streamFileToSign, "streamFileToSign must not be null");
+        Objects.requireNonNull(keyPair, "keyPair must not be null");
+
+        try {
+            final Hash entireHash = computeEntireHash(streamFileToSign.toFile());
+            final byte[] fileHashByte = entireHash.getValue();
+            final byte[] signature = sign(fileHashByte, keyPair);
+
+            generateSigBalanceFile(signatureFileDestination.toFile(), signature, fileHashByte);
+
+            System.out.println("Generated signature file: " + signatureFileDestination);
+
+            return true;
+        } catch (final SignatureException | IOException e) {
+            System.err.println("Failed to sign file " + streamFileToSign.getFileName() + ". Exception: " + e);
+            return false;
+        } catch (final InvalidKeyException | NoSuchAlgorithmException | NoSuchProviderException e) {
+            throw new RuntimeException("Irrecoverable error encountered", e);
```

Comment:
Why throw here instead of `System.err` like in the previous catch clause?   (Especially since on a quick look this will fly off the top of the stack - anyway it isn't handled by caller `SignCommand.signFile`.)

Reply:
Follow the platform's code style. Since it's RuntimeException, didn't have to catch it. @alittley

Reply:
Yes but the question is: what happens to it?  Does it fly off the top of the stack and then the user sees a stack trace with some message buried somewhere in it?  Or is it handled by some caller and changed into a `System.err.println` that the user can understand?

Reply:
You are right. We should not just throw exception. I will add System.err as I am working on the record stream sign.

Reply:
Seems to not be changed yet.  Remember it for next time.

Reply:
I fixed it and return false no one handle the throw that's why return false will be good.

Reply:
@iwsimon followed the pattern @alittley is using an in case this exception happens it is more serious and we want to throw an exception with all the stuck traces. Any issue with that?